File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> hash = new Dictionary<int,int>();
        for (int i=0;i<nums.Length;i++){
            hash[nums[i]] = i;
        }
        for (int i=0; i < nums.Length ; i++){
            int diff = target - nums[i];
            if(hash.ContainsKey(diff) && hash[diff] != i){
                return new int[] {i,hash[diff]};
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& n, int t) {
        int l=0;
        int r=n.size()-1;
        while(n[l]+n[r]!=t){
            if(n[l]+n[r]>t){
                r--;
            }
            else{
                l++;
            }
        }
        return (vector<int>){l+1,r+1};
    }
};
